Libs/MRML/Logic/vtkMRMLApplicationLogic.cxx

```diff
@@ -823,3 +824,68 @@ void vtkMRMLApplicationLogic::EditNode(vtkMRMLNode* node)
   // Observers in qSlicerCoreApplication listen for this event
   this->InvokeEvent(EditNodeEvent, node);
 }
+
+//----------------------------------------------------------------------------
+void vtkMRMLApplicationLogic::RegisterModuleLogic(const char* moduleName,
+                                                  vtkMRMLAbstractLogic* moduleLogic)
+{
+  if (!moduleName)
+    {
+    vtkErrorMacro("RegisterModuleLogic: invalid module name.");
+    return;
+    }
+
+  // NOTE: Should we accept registation of modules that have no logic (e.g. Welcome)
+  //       I think we should, and return nullptr on GetModuleLogic. This can be a good
+  //       way to indicate there is that module that has no logic, as opposed to, the module
+  //       is not registered (could be that the module does not exists at all).
+  // if (!moduleLogic)
+  //   {
+  //   vtkErrorMacro("RegisterModuleLogic: module logic.");
+  //   return;
+  //   }
+
+  // Check that the module has not any registered logic.
+  if (!this->Internal->ModuleLogicMap.empty())
+    {
+    vtkWarningMacro("RegisterModuleLogic: Logic already register for " << moduleName);
+    return;
+    }
+
+  this->Internal->ModuleLogicMap[moduleName] = moduleLogic;
+}
+
+//----------------------------------------------------------------------------
+void vtkMRMLApplicationLogic::UnregisterModuleLogic(const char* moduleName)
```

Comment:
If we rename RegisterModuleLogic to SetModuleLogic then we can remove this method.